File: code/ii-service/src/main/java/com/owlunit/service/cinema/PersonService.java
package com.owlunit.service.cinema;

import com.owlunit.model.cinema.Person;
import com.owlunit.model.cinema.Role;
import com.owlunit.service.exception.NotFoundException;

/**
 * @author Ilya Pimenov
 *         Owls Proprietary
 */
public interface PersonService {

    Person createPerson(Person person);

    Person addRole(Person person, Role role) throws NotFoundException;

    Person findOrCreate(Person person);
}

File: code/ii-service/src/main/scala/com/owlunit/service/cinema/PersonService.scala
package com.owlunit.service.cinema

import impl.PersonServiceImpl
import com.owlunit.core.ii.IiDao

/**
 * @author Anton Chebotaev
 *         Owls Proprietary
 */

class Person (override val id: Long,
              val name: String,
              val surname: String) extends CinemaItem(id) {

  def copy(id: Long) = new Person(id, this.name, this.surname)

}

trait PersonService {

  def create(sample: Person): Person
  def load(name: String, surname: String): Option[Person]
  def loadOrCreate(name: String, surname: String): Person
  def search(prefix: String): Seq[Person]

}

object PersonService {

  def apply(dao: IiDao): PersonService = new PersonServiceImpl(dao)

}
